code-goodbuy/goodbuy-nodejs
# Enhancement/logging

This mini PR is for extended version of morgan for rest api server monitoring: date/user agent/method/url/status/latency data on every request will be funneled into `src/log`. If you think more info is needed, please let me know! 

The log file will be terminated every 24h, while aws cloudwatch will pull the log to dashboard and store it into s3 on every 24h. 

Oh, Prettier might have touched some linting stuff, ignore it please! We should've consolidated prettierrc earlier I guess.. 

To test:
1. pull/checkout enhancement/logging branch
2. npm i
3. npx tsc --watch
4. npm run local
5. Try some crud action via Postman to generate log
6. Check the log in `/src/log`
7. You shoud be able to see something like this:
![image](https://user-images.githubusercontent.com/38537908/115125812-ef2f5980-9fca-11eb-9f50-62e2593ad78c.png)
